WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerAdapter.java

```diff
@@ -213,49 +218,63 @@ private boolean isValidPosition(int position) {
         return position >= 0 && position < mMediaList.size();
     }
 
-    void setMultiSelectEnabled(boolean enabled) {
-        if (mIsMultiSelectEnabled == enabled) {
-            return;
-        }
-
-        mIsMultiSelectEnabled = enabled;
+    private boolean isItemSelected(int position) {
+        return mSelectedItems.contains(position);
+    }
 
-        if (!enabled && mSelectedUris.size() > 0) {
-            mSelectedUris.clear();
-            notifyDataSetChangedInternal();
-        }
+    private void toggleItemSelected(int position) {
+        setItemSelected(position, !isItemSelected(position));
     }
 
-    /*
-     * toggles the selection state of the item at the passed position
-     */
-    private void toggleSelection(ThumbnailViewHolder holder, int position) {
+    private void setItemSelected(int position, boolean isSelected) {
         PhotoPickerItem item = getItemAtPosition(position);
         if (item == null) {
             return;
         }
 
-        boolean isSelected;
-        int selectedIndex = mSelectedUris.indexOfUri(item.mUri);
-        if (selectedIndex > -1) {
-            mSelectedUris.remove(selectedIndex);
-            isSelected = false;
-            holder.mTxtSelectionCount.setText(null);
+        if (isSelected) {
+            // if an item is already selected and multiselect isn't allowed, deselect the previous selection
+            if (!canMultiselect() && !mSelectedItems.isEmpty()) {
+                ThumbnailViewHolder prevHolder = getViewHolderAtPosition(mSelectedItems.get(0));
+                if (prevHolder != null) {
+                    AniUtils.scale(prevHolder.mImgThumbnail, SCALE_SELECTED, SCALE_NORMAL, AniUtils.Duration.SHORT);
+                }
+                mSelectedItems.clear();
+            }
+            mSelectedItems.add(position);
         } else {
-            mSelectedUris.add(item.mUri);
-            isSelected = true;
-            holder.mTxtSelectionCount.setText(String.format(Locale.getDefault(), "%d", mSelectedUris.size()));
+            int selectedIndex = mSelectedItems.indexOf(position);
+            if (selectedIndex > -1) {
+                mSelectedItems.remove(selectedIndex);
+            }
```

Comment:
Can we simplify the search+removal into just `mSelectedItems.remove((Integer) position)` perhaps?